### ISSUE TITLE
Pytorch RNNLM Without CTC

### ISSUE THREAD

**Issue by m-wiesner** [espnet/espnet#123]
When CTC weight is 0, RNNLM is not used.

**Comment by sw005320** [comment]
Thanks for the report. I'm trying to find it, but could you give me more input? Then, I could figure out what happens.

**Comment by m-wiesner** [comment]
In recognize_beam() local_scores is initially set properly. However, the next line has a condition asking whether ctc is used in decoding. If it is used, things seem fine, however if the condition evaluates false then local_scores is reset by the last line that I pasted below.

if rnnlm:
                    rnnlm_state, z_rnnlm = rnnlm.predictor(hyp['rnnlm_prev'], vy)
                    local_lm_scores = F.log_softmax(z_rnnlm, dim=1).data
                    local_scores = local_att_scores + recog_args.lm_weight * local_lm_scores
                else:
                    local_scores = local_att_scores


if lpz is not None:
....
else:
       local_best_scores, local_best_ids = torch.topk(local_att_scores, beam, dim=1)

**Comment by shrutijpalaskar** [comment]
I've observed the same thing

**Comment by sw005320** [comment]
Just make sure. This only happens in the pytorch, right? I think I can fix it.

**Comment by sw005320** [comment]
Can you test it?
We've finished #126

**Comment by sw005320** [comment]
This seems to be solved.
